component/otelcol/processor/batch/batch.go

```diff
@@ -0,0 +1,87 @@
+// Package batch provides an otelcol.processor.batch component.
+package batch
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/processor"
+	"github.com/grafana/agent/pkg/river"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelconfig "go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/processor/batchprocessor"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "otelcol.processor.batch",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := batchprocessor.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.processor.batch component.
+type Arguments struct {
+	Timeout          time.Duration `river:"timeout,attr,optional"`
+	SendBatchSize    uint32        `river:"send_batch_size,attr,optional"`
+	SendBatchMaxSize uint32        `river:"send_batch_max_size,attr,optional"`
+
+	// Output configures where to send processed data. Required.
+	Output *otelcol.ConsumerArguments `river:"output,block"`
```

Comment:
Do we want to reuse the `forward_to` terminology we used for Prometheus, or is this more in-line with what OTel does?

Reply:
Maybe, I think we need to think about it after we have a good set of otelcol components in. It might make sense for us to use different terminology just to help prevent confusing things (i.e., assume that you can use forward_to to send data to a `prometheus.*` receiver)